azurerm/internal/services/desktopvirtualization/tests/virtual_desktop_workspace_resource_test.go

```diff
@@ -0,0 +1,189 @@
+				ExpectError: acceptance.RequiresImportError("azurerm_virtual_desktop_workspace"),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMDesktopVirtualizationWorkspaceExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).DesktopVirtualization.WorkspacesClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		id, err := parse.DesktopVirtualizationWorkspaceID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		result, err := client.Get(ctx, id.ResourceGroup, id.Name)
+
+		if err == nil {
+			return nil
+		}
+
+		if result.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: Virtual Desktop Workspace %q (Resource Group: %q) does not exist", id.Name, id.ResourceGroup)
+		}
+
+		return fmt.Errorf("Bad: Get virtualDesktopWorspaceClient: %+v", err)
+	}
+}
+
+func testCheckAzureRMDesktopVirtualizationWorkspaceDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).DesktopVirtualization.WorkspacesClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_virtual_desktop_workspace" {
+			continue
+		}
+
+		log.Printf("[WARN] azurerm_virtual_desktop_workspace still exists in state file.")
+
+		id, err := parse.DesktopVirtualizationWorkspaceID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		result, err := client.Get(ctx, id.ResourceGroup, id.Name)
+
+		if err == nil {
+			return fmt.Errorf("Virtual Desktop Workspace still exists:\n%#v", result)
+		}
+
+		if result.StatusCode != http.StatusNotFound {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMDesktopVirtualizationWorkspace_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_desktop_workspace" "test" {
+  name                = "acctws%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+}
+
+func testAccAzureRMDesktopVirtualizationWorkspace_complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_desktop_workspace" "test" {
+  name                = "acctws%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  friendly_name       = "acceptance test"
```

Comment:
capitals when possible to ensure the API handles them correctly
```suggestion
  friendly_name       = "Acceptance Test!"
```